pkg/api/service/scans.go

```diff
@@ -136,38 +68,49 @@ type scanStats struct {
 	NumberOfChecksPerChecktype map[string]int
 }
 
+// ChecktypesByAssettypes is used as a lookup table to check if a checktype can
+// be run against a concrete assettype.
+type ChecktypesByAssettypes map[string]map[string]struct{}
+
 // ScansService implements the functionality needed to create and query scans.
 type ScansService struct {
-	db            ScansPersistence
-	logger        log.Logger
-	vulcanCore    CoreAPI
-	notifier      ScanNotifier
-	ccreator      *CheckCreator
-	metricsClient metrics.Client
-	accreator     AsyncChecksCreator
+	db             persistence.ScansStore
```

Comment:
Just curious, why adding a reference to an interface defined in the persistence instead of declaring the interface, with only the methods needed, in the ScansService?.  I'm not saying is something bad or wrong by its own just to know the reasons behind the decision.

Reply:
Yes, I mean, for me I think that pattern of defining a "smaller subset" interface with just the methods you need works well when you want to abstract a "component" that exports many more methods than you want to consume, so you don't have to know about them nor have access.
But in this case I think both service and persistence are very tight together because if persistence interface exposes a method is most probably because the service layer needs it to do its operations so I in this cases I think its best to use the interface directly instead of created more "intermediate interfaces" that also add a little more complexity when following the code.

What do you think?

Reply:
I agree with you in this case thanks for the explanation!!